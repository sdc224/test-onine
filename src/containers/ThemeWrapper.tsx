import React from "react";
import { ThemeProvider, ColorModeProvider, CSSReset } from "@chakra-ui/core";

interface ThemeWrapperProps {
	children: React.ReactNode;
}

export default function ThemeWrapper({ children }: ThemeWrapperProps) {
	return (
		<ThemeProvider>
			<CSSReset />
			<ColorModeProvider>{children}</ColorModeProvider>
		</ThemeProvider>
	);
}
