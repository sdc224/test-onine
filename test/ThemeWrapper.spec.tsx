import React from "react";
import { render, RenderResult } from "@testing-library/react";
import ThemeWrapper from "@containers/ThemeWrapper";
import MockComponent from "./mocks/MockComponent";

let documentBody: RenderResult;

describe("<ThemeWrapper />", () => {
	beforeEach(() => {
		documentBody = render(
			<ThemeWrapper>
				<MockComponent />
			</ThemeWrapper>
		);
	});
	it("matches snapshot", () => {
		const { baseElement } = documentBody;
		expect(baseElement).toMatchSnapshot();
	});
});
