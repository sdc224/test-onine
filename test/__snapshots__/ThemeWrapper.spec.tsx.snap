// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeWrapper /> matches snapshot 1`] = `
<body
  class="light-mode"
>
  <div
    class="Toaster"
    id="react-toast"
  >
    <span
      class="Toaster__manager-top"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; top: 0px; right: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-top-left"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; top: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-top-right"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; top: 0px; right: 0px;"
    />
    <span
      class="Toaster__manager-bottom-left"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-bottom"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; margin: 0px auto; text-align: center; bottom: 0px; right: 0px; left: 0px;"
    />
    <span
      class="Toaster__manager-bottom-right"
      style="max-width: 560px; position: fixed; z-index: 5500; pointer-events: none; bottom: 0px; right: 0px;"
    />
  </div>
  <div>
    <p>
      Mock Component
    </p>
  </div>
</body>
`;
